collect_app/src/main/java/org/odk/collect/android/tasks/sms/models/Message.java

```diff
@@ -0,0 +1,54 @@
+package org.odk.collect.android.tasks.sms.models;
+
+import java.util.Random;
+
+/**
+ * Represents a single message that's been sent.
+ */
+public class Message {
+    private int part;
+    private int id;
+    private String text;
+    private MessageStatus messageStatus;
+
+    public int getPart() {
+        return part;
+    }
+
+
+    public int getId() {
+        return id;
+    }
+
+    public void setPart(int part) {
+        this.part = part;
+    }
+
+    public String getText() {
+        return text;
+    }
+
+    public void setText(String text) {
+        this.text = text;
+    }
+
+    public boolean isSent() {
+        return messageStatus.equals(MessageStatus.Sent);
+    }
+
+    public boolean isSending() {
+        return messageStatus.equals(MessageStatus.Sending);
+    }
+
+    public void generateRandomMessageID() {
+        id = new Random().nextInt(10000);
```

Comment:
Nice explanation! I didn't use an uuid because I am using the message id as the unique identifier for the `PendingIntent` being created for each message and that parameter is an integer.

Reply:
You still need some way to avoid collisions. If you have some structure keeping track of ids (e.g. a map from ids to messages in the manager), you can keep generating values until you get a unique one.

Reply:
So the proper approach here would be to simply create functionality that does a traversal of all submission messages to check the uniqueness of a generated id correct?

Reply:
Yes, something like that. I was thinking you could do that when you build your `SmsSubmission`s but the ID has to be unique across possibly multiple `SmsSubmission`s.

Reply:
Hmm, this is going to be a challenge because I don't think you have an in-memory representation of all ongoing submissions, you always read them from prefs, right?

Reply:
Yes, I always have to do that.

Reply:
Well, maybe for now you can make the upper bound for `nextInt` `Integer.MAX_VALUE` which reduces the risk of collisions significantly and we can revisit if constantly reading/writing from prefs causes other problems. Would it have been possible to keep things in memory while the activity is running and only serialize when the activity was going away?

Reply:
Sure. I could look into doing that so I would add methods to the `SubmissionManager` such as `activityAttched` and `activityDetached` that would trigger this behaviour right? And then from there it would be easier for me to transfer the submissions since they are in memory and only when they the app is background during the id generation process should I fetch the prefs?